modules/speech-to-text-async-impl/src/main/java/org/opencastproject/speechtotext/async/impl/SpeechToTextWorkflowSchedulerQuartz.java

```diff
@@ -0,0 +1,403 @@
+import org.quartz.impl.StdSchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Quartz job to start workflows to asynchronously attach captions generated by the SpeechToText service.
+ */
+@Component(property = { "service.description=Speech to Text Workflow Scheduler" }, immediate = true, service = {
+        ManagedService.class })
+public class SpeechToTextWorkflowSchedulerQuartz extends AbstractScanner implements ManagedService {
+  private static final Logger logger = LoggerFactory.getLogger(SpeechToTextWorkflowSchedulerQuartz.class);
+
+  public static final String JOB_GROUP = "org-opencast-stt-wf-scheduler-group";
+  public static final String JOB_NAME = "org-opencast-stt-wf-scheduler-job";
+  public static final String SCANNER_NAME = "STT Worflow Scheduler Quartz";
+  public static final String TRIGGER_GROUP = "org-opencast-stt-wf-scheduler-trigger-group";
+  public static final String TRIGGER_NAME = "org-opencast-stt-wf-scheduler-trigger";
+
+  // === Configuration options
+  /* Number of seconds to abandon retrying attaching subtitles after they are generated. */
+  public static final String ABANDON_AFTER_SECS = "abandon-after-secs";
+  /* Workflow definition to use to attach the subtitles */
+  public static final String WORKFLOW = "workflow";
+
+  // === Default values
+  /* Default number of seconds to abandon retrying attaching subtitles. */
+  private static final long DEFAULT_ABANDON_AFTER_SECS = 24 * 60 * 60;
+
+  /* Workflow definition to start to attach the subtitles */
+  private String attachWorkflowDef;
+  /* If STT jobs not finished after this interval, they won't be tracked anymore. */
+  private long abandonAfterMs = DEFAULT_ABANDON_AFTER_SECS * 1000;
+
+  private AssetManager assetManager;
+  private WorkflowService workflowService;
+  private SpeechToTextDatabase database;
+
+  // Only used by unit tests!
+  private Workflows wfUtil;
+
+  @Override
+  @Activate
+  protected void activate(ComponentContext cc) {
+    logger.info("Activating!");
+    super.activate(cc);
+  }
+
+  public SpeechToTextWorkflowSchedulerQuartz() {
+    try {
+      quartz = new StdSchedulerFactory().getScheduler();
+      quartz.start();
+      // create and set the job. To actually run it call schedule(..)
+      final JobDetail job = new JobDetail(getJobName(), getJobGroup(), Runner.class);
+      job.setDurability(false);
+      job.setVolatility(true);
+      job.getJobDataMap().put(JOB_PARAM_PARENT, this);
+      quartz.addJob(job, true);
+    } catch (org.quartz.SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
```

Comment:
Because we're not doing `ManagedService`
```suggestion
  @Modified
  public void modified(Map<String, Object> properties) {
```